### ISSUE TITLE
Fix issue with delete_cluster_buckets

### ISSUE THREAD

**Issue by clacroix12** [red-hat-storage/ocs-ci#2604]
From #2549 

```    
        if http.status_code >= 300:
            error_code = parsed_response.get("Error", {}).get("Code")
            error_class = self.exceptions.from_code(error_code)
>           raise error_class(parsed_response, operation_name)
E           botocore.errorfactory.NoSuchBucket: An error occurred (NoSuchBucket) when calling the ListObjects operation: The specified bucket does not exist

.venv/lib/python3.6/site-packages/botocore/client.py:635: NoSuchBucket
=============================== warnings summary ===============================
```

I don't think we should fail the upgrade if bucket not found but just continue here.

So re-opening this.

Job:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-destroy-ocs-cluster/7945/console

This is actually an issue with `delete_cluster_buckets` that was added [in this PR](https://github.com/red-hat-storage/ocs-ci/pull/2420/files)

**Comment by clacroix12** [comment]
@petr-balogh The only scenario I could reproduce this issue in is when we found the bucket name initially but the bucket did not exist when we went to delete the files/bucket. In this scenario we don't want to error out and stop further teardown as the bucket is already non-existant. I have added a try/except in and we simply log any ClientErrors that we encounter when trying to delete those buckets.
